implementing_installation.md

```diff
@@ -0,0 +1,41 @@
+# Implementing Installation Information
+
+This document exists as a guide for launchers and other automation tools that may work with installation information. This is not a comprehensive manual and it may not fit for every launcher. However, it's recommended to follow this guide as accurately as possible in order to have a smooth experience.
+
+## Implementing `placeInDirectory`
+
+The `directory` field in this value should be relative to Minecraft installation that the project is being installed to. This field is subject to path rules defined in **installation.md - Specifying Paths**. Launchers should check for rule violations in any field where a directory is passed in order to ensure format-compliant information.
+
+Files should be placed with the same filename and file extension as when downloaded.
+
+## Implementing `versionJsonInstall`
+
+Launchers will likely already have the mechanisms in place to parse Minecraft version JSON files. To install a Version with the `versionJsonInstall` method, first find a file with the `rel` value set to `versionJson`. If a file is not present, an error should be presented to the user. If multiple files are found, the first one should be used.
+
+Launchers can then use the file with the `versionJson` relation and automatically install libraries from URLs specified in the JSON file.
```

Comment:
What format is the file?

Also I think all the metadata should be in the index file - we shouldn't have files that contain more metadata. Don't know the best way to implement that. Maybe `classpathLibrary` with an additional Maven address on the primary file?